lib/azure/resource_management/ARM_interface.rb

```diff
@@ -29,17 +29,17 @@ class ARMInterface < AzureInterface
       include Azure::ARM::ARMDeploymentTemplate
       include Azure::ARM::VnetConfig
 
-      include Azure::ARM::Resources
-      include Azure::ARM::Resources::Models
+      include Azure::Resources::Mgmt::V2018_05_01
+      include Azure::Resources::Mgmt::V2018_05_01::Models
 
-      include Azure::ARM::Compute
-      include Azure::ARM::Compute::Models
+      include Azure::Compute::Mgmt::V2018_06_01
+      include Azure::Compute::Mgmt::V2018_06_01::Models
 
-      include Azure::ARM::Storage
-      include Azure::ARM::Storage::Models
+      include Azure::Storage::Mgmt::V2018_07_01
+      include Azure::Storage::Mgmt::V2018_07_01::Models
 
-      include Azure::ARM::Network
-      include Azure::ARM::Network::Models
+      include Azure::Network::Mgmt::V2018_08_01
+      include Azure::Network::Mgmt::V2018_08_01::Models

```

Comment:
It looks like this wasn't added until https://github.com/Azure/azure-sdk-for-ruby/commit/8fb678ddaad195b1da2d31b073d82379b000cc66 which means your dependency on `azure_mgmt_network` should be `"~> 0.17.3"

https://rubygems.org/gems/azure_mgmt_network/versions/0.17.3